rudder-core/src/main/scala/com/normation/rudder/services/path/PathComputerImpl.scala

```diff
@@ -57,25 +57,32 @@ class PathComputerImpl(
 ) extends PathComputer with Loggable {
 
 
-  private[this] val promisesPrefix = "rules/"
+  private[this] val promisesPrefix = "/rules"
+  private[this] val newPostfix = ".new"
+  private[this] val backupPostfix = ".bkp"
 
   private[this] val baseFolder = Constants.NODE_PROMISES_PARENT_DIR_BASE
   private[this] val relativeShareFolder = Constants.NODE_PROMISES_PARENT_DIR
 
   /**
-   * Compute the base path for a machine, i.e. the full path on the root server to the data
-   * the searched machine will fetch, and the backup folder
-   * Finish by the machine uuid, with no trailing /
-   * It does not contain the root machine, except if we search the root machine
-   * Ex : /var/rudder/share/uuid-a/share/uuid-b, /var/rudder/backup/uuid-a/share/uuid-b
+   * Compute
+   *  the path of promises for a node, i.e. the full path on the root server to the data
+   *  the new folder, i.e. the full path where promises are written and checked before being updated
+   *  the backup folder, the path were promises are backuped
+   * Finish with no trailing /
+   * It does not contain the root machine, and does not work for root
+   * Ex : /var/rudder/share/uuid-a/share/uuid-b/rules,/var/rudder/share/uuid-a/share/uuid-b/rules.new, /var/rudder/backup/uuid-a/share/uuid-b
    * @param searchedNodeConfiguration : the machine we search
    * @return
    */
-  def computeBaseNodePath(searchedNodeId : NodeId, rootNodeId: NodeId, allNodeConfigs: Map[NodeId, NodeConfiguration]): Box[((String, String))] = {
+  def computeBaseNodePath(searchedNodeId : NodeId, rootNodeId: NodeId, allNodeConfigs: Map[NodeId, NodeConfiguration]): Box[((String, String, String))] = {
     for {
       path <- recurseComputePath(rootNodeId, searchedNodeId, "/"  + searchedNodeId.value, allNodeConfigs)
```

Comment:
In fact, it's more the comment that we need to change:
"It does not contain the root machine, and does not work for root"
"The path calculated for the root server is not the one expected by cfengine and so must not be used. A special case must be done for root server where the method is called".